BPA/Ch.07/cjs_mnl_ran.stan

```diff
@@ -24,47 +24,50 @@ transformed parameters {
   vector<lower=0,upper=1>[n_occasions - 1] q;
   simplex[n_occasions] pr[n_occasions - 1];
 
-  mu <- logit(mean_phi);
+  mu = logit(mean_phi);
 
   // Constraints
   for (t in 1:(n_occasions - 1)) {
-    phi[t] <- inv_logit(mu + epsilon[t]);
-    p[t] <- mean_p;
+    phi[t] = inv_logit(mu + epsilon[t]);
+    p[t] = mean_p;
   }
 
-  q <- 1.0 - p;             // Probability of non-recapture
+  q = 1.0 - p;             // Probability of non-recapture
 
   // Define the cell probabilities of the m-arrays
   // Main diagonal
   for (t in 1:(n_occasions - 1)) {
-    pr[t][t] <- phi[t] * p[t];
+    pr[t][t] = phi[t] * p[t];
 
     // Above main diagonal
     for (j in (t + 1):(n_occasions - 1))
-       pr[t][j] <- prod(segment(phi, t, j - t + 1)) *
-                   prod(segment(q, t, j - t)) *
-                   p[j];
+      pr[t][j] = prod(phi[t:j])
+               * prod(q[t:(j - 1)])
+               * p[j];
 
     // Below main diagonal
     for (j in 1:(t - 1))
-      pr[t][j] <- 0;
-  } // t
+      pr[t][j] = 0;
+  }
 
   // Last column: probability of non-recapture
   for (t in 1:(n_occasions - 1))
-    pr[t][n_occasions] <- 1 - sum(head(pr[t], n_occasions - 1));
+    pr[t][n_occasions] = 1 - sum(pr[t][1:(n_occasions - 1)]);
```

Comment:
You can have this be `pr[t, 1:(n_occasions - 1)]`.

Given how often you use this `n_occasions - 1` calculation, you might want to make a transformed data item and then just reuse it.